### PR TITLE
NOTES.WIN: classify targets to "native" and "hosted" and restructure.

### PR DESCRIPTION
**Pull request by dot-asm**
[skip ci]

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: NOTES.WIN [hunk 1]
```diff
@@ -2,8 +2,35 @@
...
+ start, and if it's found *not* to be linked with MSYS2 POSIX-y thing,
+ it translates command line arguments that look like file names from
+ emulated name space to "native". For example '/c/some/where' becomes
+ 'c:\some\where', '/dev/null' - 'nul'. This creates an illusion that
+ there is no difference between MSYS2 shell and "MinGW binary", but
+ there is. Just keep in mind that "MinGW binary" "experiences" Windows
+ system in exactly same way as one produced by VC, and in its essence
+ is indistinguishable from the latter. [Which by the way is why
+ it's referred to in quotes here, as "MinGW binary", it's just as
+ "native" as it can get.]
```
Comment (levitte) [inline NOTES.WIN]: Oh!  There's one thing I've notice in your style of writing that comes out as ...  odd.  You use squares parentheses [such as this] when one would usually use round parentheses (such as these).

It becomes a bit extra confusing when you have things like `mingw[64]` and `MSYS[2]`, where the part in square brackets are optional.

Would you mind terribly following the norm re the use of parentheses in written natural languages (or more specifically, English)?  Note that I'm only asking this for things that *are* natural language (so no one thinks I'm trying to say that `mingw[64]` should be changed to `mingw(64)`...  I'm certainly not saying that!)

File: NOTES.WIN [hunk 2]
```diff
@@ -55,62 +82,36 @@
...
- with "conventional" Windows binaries you generate with/for MinGW.
+ mingw and mingw64
+ =================
 
+ * MSYS shell and development environment installation:
 
- GNU C (MinGW/MSYS)
- ------------------
+   Download MSYS2 from https://msys2.github.io/ and follow installation
+   instructions. Once up and runinng install even make, perl, [git,]
```
Comment (levitte) [inline NOTES.WIN]: s/runinng/running/

File: NOTES.WIN (hunk 1 above)
Comment (dot-asm) [reply NOTES.WIN]: I suppose I mechanically extend that "manual page" convention, when [] denote something optional, or something that can be omitted, to sentences. But sure, I'll fix. (As more feedback provided. And if not, then just fix after some grace period.) Like this, right?

File: NOTES.WIN (hunk 2 above)
Comment (dot-asm) [reply NOTES.WIN]: BTW, what do you think of [git,] here?

File: NOTES.WIN (hunk 1 above)
Comment (levitte) [reply NOTES.WIN]: (yes, exactly like that)

File: NOTES.WIN (hunk 2 above)
Comment (levitte) [reply NOTES.WIN]: I would make that `(git if you need it)`

File: NOTES.WIN [hunk 3]
```diff
@@ -35,17 +35,17 @@
...
    You also need the perl module Text::Template, available on CPAN.
    Please read NOTES.PERL for more information.
 
- - You need a Microsoft Visual C compiler. There is ambiguity in which
-   versions are supported. It's probably safe to say that latest along
-   with couple of previous are fully supported. On the other hand we
-   know for sure that eldest one does not work. Everything between falls
-   to best-effort category.
+ - Microsoft Visual C compiler. Since we can't test them all, there is
+   natural uncertainty in which versions are supported. It's probably
```
Comment (levitte) [inline NOTES.WIN]: s/in which/what/

Comment (levitte) [reply NOTES.WIN]: Also, I'm uncertain about "there is natural uncertainty"...  let me ponder, it's currently late at night.

File: NOTES.WIN [hunk 4]
```diff
@@ -85,15 +85,15 @@
...
-   corresponding MinGW items on your start menu, use *them*, not generic
-   MSYS2. As implied in opening note, difference between them is which
-   compiler is found 1st on $PATH. At this point ./config should recognize
-   correct target...
+   instructions. Once up and running install even make, perl, (git if
+   need it,) mingw-w64-i686-gcc and/or mingw-w64-x86_64-gcc. You should
+   have corresponding MinGW items on your start menu, use *them*, not
+   generic MSYS2. As implied in opening note, difference between them
+   is which compiler is found 1st on $PATH. At this point ./config
+   should recognize correct target, roll as it was Unix...
```
Comment (levitte) [inline NOTES.WIN]: s/as it/as if it/

File: NOTES.WIN (hunk 3 above)
Comment (dot-asm) [reply NOTES.WIN]: I can't agree with "s/in which/what". "What" would work in "what is supported", but we are talking about a subset of multiple versions, and you can't say "what one". I've also rephrased "natural uncertainty in" as "unavoidable uncertainty about". Does it sound better? What I want to achieve is to say that we effectively can't answer overly specific questions about compiler versions.

File: NOTES.WIN (hunk 4 above)
Comment (dot-asm) [reply NOTES.WIN]: Fixed.

File: NOTES.WIN [hunk 5]
```diff
@@ -2,32 +2,59 @@
...
+ Requirement details
+ -------------------
 
  In addition to the requirements and instructions listed in INSTALL,
  these are required as well:
 
- - You need Perl.  We recommend ActiveState Perl, available from
+ - Perl. We recommend ActiveState Perl, available from
    https://www.activestate.com/ActivePerl. Another viable alternative
    appears to be Strawberry Perl, http://strawberryperl.com.
```
Comment (levitte) [inline NOTES.WIN]: Side note: Strawberry Perl seems to do some weird asynchronous subprocess when `exec` is used.  When I do `perl configdata.pm -r` with it, I get the cmd prompt back immediately, and then I start getting configuration output.  As far as I've seen, this doesn't happen with ActiveState perl...

Comment (dot-asm) [reply NOTES.WIN]: Well, wording allows for quirks like that, as Strawberry is described as *apparently* viable alternative ;-) But it does produce working build if you stay on most common track. I actually use it for my test builds...

**Comment by dot-asm** [comment]
Minor typo fixed. Since it's non-code change, and minor typo, I consider that approval stands [unless somebody else claims otherwise].

**Pull request closed without merging**